### PR TITLE
PEP 617: Add section on rejected alternatives

### PR DESCRIPTION
**Pull request by gvanrossum**
Closes https://github.com/we-like-parsers/cpython/issues/34

### CODE REVIEW

**Comment by gvanrossum** [comment]
CC: @lysnikolaou

File: pep-0617.rst [hunk 1]
```diff
@@ -755,6 +755,31 @@ faster but uses slightly (about 10%) more memory. We believe this is
...
+Rejected Alternatives
+=====================
+
+We did not seriously consider alternative ways to implement the new
+parser, but here's a brief discussion of LALR(1).
+
+Thirty years ago the first author decided to go his own way with
+Python's parser rather than using LALR(1), which was the industry
+standard at the time (e.g. Bison and Yacc).  The reasons were
+primarily emotional (gut feelings, intuiton), based on past experience
```
Comment (pablogsal) [inline pep-0617.rst:768]:
```suggestion
primarily emotional (gut feelings, intuition), based on past experience
```

File: pep-0617.rst [hunk 2]
```diff
@@ -755,6 +755,31 @@ faster but uses slightly (about 10%) more memory. We believe this is
...
+using Yacc in other projects, where grammar development took more
+effort than anticipated (in part due to shift-reduce conflicts).  A
+specific criticism of Bison and Yacc that still holds is that their
+meta-grammar (the notation used to feed the grammar into the parser
+generator) does not support EBNF conveniences like
+``[optional_clause]`` or ``(repeated_clause)*``.  Using a custom
+parser generator a syntax tree matching the structure of the grammar
+could be generated automatically, and with EBNF that tree could match
+the "human-friendly" structure of the grammar.
```
Comment (pablogsal) [inline pep-0617.rst:777]:
```suggestion
parser generator, a syntax tree matching the structure of the grammar
could be generated automatically, and with EBNF that tree could match
the "human-friendly" structure of the grammar.
```

Comment (pablogsal) [reply pep-0617.rst:777]: Not sure about if the comma there is correct, but I had problems reading this sentence the first time.

Comment (lysnikolaou) [reply pep-0617.rst:777]: Exactly what I was going to say. I'm not sure, where the comma should go, but I feel there is some missing. I still have to go through the sentence more than once to completely follow it.

**Pull request merged**